### PR TITLE
TASK-8: Setup Media Backend

### PR DESCRIPTION
**Pull request by leojrsilao**
**Task Description**
This PR addresses #36.

**Note about GraphQL decorators and code coverage**
If you see ```/* istanbul ignore next */``` on any of the GraphQL classes, it is to ignore them from the code coverage since the decorators take in anonymous functions.  

**Testing Steps Using Postman**
1. Make sure to resolve all your dependencies with `npm i`
1. Run the application using ```nx serve mull-api```.
1. Open the Postman application
1. Select POST from the dropdown
1. Enter URL to GraphQL server `http://localhost:3333/graphql`
1. Select the `Body` tab and `form-data`
1. Fill out the `Body` tab as follows:

| Key | Value |
| ------------- | ------------- |
| operations | {"query":"mutation UploadFile($file: Upload!) {uploadFile(file:$file)}", "variables": {"file": null}}  |
| map | {"0": ["variables.file"]}  |
| 0 | [File ▾] (select your file to upload) |

8. Observe that the image is uploaded within the `apps/mull-api/uploads` folder and stored in the database

### CODE REVIEW

File: apps/mull-api/src/app/app.module.ts
Comment (cristian-aldea) [inline apps/mull-api/src/app/app.module.ts:17]: You moved this import, but you didn't need to.

File: apps/mull-api/src/app/app.module.ts
Comment (cristian-aldea) [inline apps/mull-api/src/app/app.module.ts:8]: You didn't need to remove this comment

File: apps/mull-api/src/app/media/media.resolver.spec.ts
Comment (cristian-aldea) [inline apps/mull-api/src/app/media/media.resolver.spec.ts]: You have a try-catch where after catching an error, you throw it right back. You don't even need the try-catch in that case.
```suggestion
        renameSync(
          join(process.cwd(), `apps/mull-api/uploads/${mockPrevFilename}`),
          join(process.cwd(), `apps/mull-api/uploads/${mockNextFilename}.${mockFileType}`)
```

File: apps/mull-api/src/app/app.module.ts
Comment (cristian-aldea) [inline apps/mull-api/src/app/app.module.ts]: You can move these to the top of the imports array, where UserModule was originally.

File: apps/mull-api/src/app/media/media.resolver.spec.ts
Comment (cristian-aldea) [inline apps/mull-api/src/app/media/media.resolver.spec.ts]: You can change it to this if you want:
```suggestion
  it('should not upload an invalid file', async () => {
```

File: apps/mull-api/src/app/media/media.service.spec.ts
Comment (cristian-aldea) [inline apps/mull-api/src/app/media/media.service.spec.ts]: You can rename it to this if you want:
```suggestion
  it('should not create an invalid file', async () => {
```

File: apps/mull-api/src/app/media/media.service.spec.ts
Comment (cristian-aldea) [inline apps/mull-api/src/app/media/media.service.spec.ts]: You can rename it to this to be more specific:
```suggestion
  it('should not rename a nonexistent file', async () => {
```

File: apps/mull-api/src/app/media/media.service.ts
Comment (cristian-aldea) [inline apps/mull-api/src/app/media/media.service.ts]: Same as above, you don't need a try/catch.
```suggestion
      renameSync(
        join(process.cwd(), `apps/mull-api/uploads/${prevFilename}`),
        join(process.cwd(), `apps/mull-api/uploads/${nextFilename}.${fileType}`)
      );
```

File: apps/mull-api/src/schema.gql
Comment (cristian-aldea) [inline apps/mull-api/src/schema.gql]: Add the following to the `.prettierignore` file:
```
schema.gql
```
And then run the backend to revert the fix prettier did.

File: package.json
Comment (cristian-aldea) [inline package.json]: You don't seem to be using this package, you can remove it.
```suggestion
```

**Review by cristian-aldea — DISMISSED** [review]
Great Work! Just had a few minor comments that I hope you can address, not gonna block though.

File: package-lock.json
Comment (cristian-aldea) [inline package-lock.json]: Thanks so much for cleaning up the package, just make sure to run `npm i` and commit the changes before merging the PR, otherwise the package will still be in the package-lock.json

**Review by cristian-aldea — DISMISSED** [review]
Thanks Leo! Had one last small comment but otherwise LGTM

**Review by cristian-aldea — DISMISSED** [review]

**Review by cristian-aldea — DISMISSED** [review]

**Review by cindyslittleplanet — DISMISSED** [review]
**Acceptance Criteria**

- [x] Media backend is set up and functional
- [x] Can upload and download images
- [x] Can be reachable through a backend endpoint
- [x] Documentation present for every new component and function

LGTM!

**Comment by bit172** [comment]
Could you add the #36 on the last commit?

**Review by cindyslittleplanet — DISMISSED** [review]

**Review by cristian-aldea — DISMISSED** [review]

**Comment by leojrsilao** [comment]
Sorry that I keep dismissing reviews @TheGreatMarkus @cindyslittleplanet, had to rebase to get the new changes on develop 🤖

File: apps/mull-api/src/app/app.module.ts
Comment (ghiobi) [inline apps/mull-api/src/app/app.module.ts]: spelling

File: package.json
Comment (ghiobi) [inline package.json]: what is this package? isn't the integrated package enough?

Comment (cristian-aldea) [reply package.json]: Oh I don't think we were aware of an existing package

**Review by cristian-aldea — APPROVED** [review]

**Review by cindyslittleplanet — APPROVED** [review]

**Pull request merged**
